db/migrations/20201007080800_create_time_auction_totals_query.sql

```diff
@@ -0,0 +1,251 @@
+-- +goose Up
+CREATE TYPE api.time_bid_total AS
+(
+    bucket_start TIMESTAMP,
+    bucket_end TIMESTAMP,
+    bucket_interval INTERVAL,
+    count BIGINT,
+    lot_start NUMERIC,
+    lot_end NUMERIC,
+    bid_amount_start NUMERIC,
+    bid_amount_end NUMERIC
+);
+
+-- +goose StatementBegin
+CREATE FUNCTION api.time_flip_bid_totals(ilk_identifier TEXT, range_start TIMESTAMP, range_end TIMESTAMP, bucket_interval INTERVAL DEFAULT '1 day'::INTERVAL)
+    RETURNS SETOF api.time_bid_total AS
+$$
+DECLARE
+    r api.time_bid_total%rowtype;
+BEGIN
+    ASSERT EXTRACT(EPOCH FROM (range_end - range_start)) / EXTRACT(EPOCH FROM bucket_interval) <= 100, 'Please limit requests to at most 100 buckets.';
+    
+    FOR r IN 
+        WITH buckets AS (
+            SELECT generate_series(range_start, range_end - bucket_interval, bucket_interval) AS bucket_start,
+            extract(epoch FROM generate_series(range_start, range_end - bucket_interval, bucket_interval)) AS bucket_start_epoch,
+            extract(epoch FROM generate_series(range_start + bucket_interval, range_end, bucket_interval)) AS bucket_end_epoch
```

Comment:
We're extracting one entry from each entry in the series. So the result looks something like:

| bucket start | bucket start epoch | bucket end epoch |
| "timestamp start | "timestamp start as epoch" | "timestamp end as epoch"|
| "timestamp 2 start | "timestamp 2 start as epoch" | "timestamp 2 end as epoch"|

Keeping this note here since I had to go run it in pgadmin to be sure myself.